pkg/kubelet/eviction/threshold_notifier_linux.go

```diff
@@ -46,6 +47,14 @@ var _ CgroupNotifier = &linuxCgroupNotifier{}
 // NewCgroupNotifier returns a linuxCgroupNotifier, which performs cgroup control operations required
 // to receive notifications from the cgroup when the threshold is crossed in either direction.
 func NewCgroupNotifier(path, attribute string, threshold int64) (CgroupNotifier, error) {
+	// cgroupv2 does not support monitoring cgroup memory thresholds using cgroup.event_control.
+	// Instead long term, on cgroupv2 kubelet should rely on combining usage of memory.low on root pods cgroup with inotify notifications on memory.events and or PSI pressure.
+	// For now, let's return a fake "disabled" cgroup notifier on cgroupv2.
+	// https://github.com/kubernetes/kubernetes/issues/106331
+	if libcontainercgroups.IsCgroup2UnifiedMode() {
```

Comment:
Should we not start the notifier at all in eviction_manager while enable kernelMemcgNotification 
 but on cgroupv2 ?https://github.com/kubernetes/kubernetes/blob/c98b388a847f84019609f6422b4ab00b89c4603c/pkg/kubelet/eviction/eviction_manager.go#L184

Reply:
that would work too, but I choose to disable it in the `NewCgroupNotifier`, so calling `Start()` will be a no-op. I guess there's not big difference on where exactly to disable it.